src/plugins/EnergyRAPL.cpp

```diff
@@ -0,0 +1,154 @@
+#include "EnergyRAPL.h"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+#ifdef ENABLE_MPI
+#include <map>
+#endif
+
+#include "../utils/Logger.h"
+#include "utils/xmlfileUnits.h"
+
+EnergyRAPL::RAPLCounter::RAPLCounter(const std::string domainBasePath) {
```

Comment:
Either pass by reference
```suggestion
EnergyRAPL::RAPLCounter::RAPLCounter(const std::string &domainBasePath) {
```
Or allow move and actually move it
```suggestion
EnergyRAPL::RAPLCounter::RAPLCounter(std::string domainBasePath) {
```
I guess the first option is better suited here?